src/services/classifierVscode.ts

```diff
@@ -0,0 +1,246 @@
+          if (symbol.flags & SymbolFlags.Alias) {
+            symbol = typeChecker.getAliasedSymbol(symbol);
+          }
+          let typeIdx = classifySymbol(symbol, getMeaningFromLocation(node));
+          if (typeIdx !== undefined) {
+            let modifierSet = 0;
+            if (node.parent) {
+              const parentIsDeclaration = (isBindingElement(node.parent) || tokenFromDeclarationMapping[node.parent.kind] === typeIdx);
+              if (parentIsDeclaration && (<NamedDeclaration>node.parent).name === node) {
+                modifierSet = 1 << TokenModifier.declaration;
+              }
+            }
+
+            // property declaration in constructor
+            if (typeIdx === TokenType.parameter && isRightSideOfQualifiedNameOrPropertyAccess(node)) {
+              typeIdx = TokenType.property;
+            }
+
+            typeIdx = reclassifyByType(typeChecker, node, typeIdx);
+
+            const decl = symbol.valueDeclaration;
+            if (decl) {
+              const modifiers = getCombinedModifierFlags(decl);
+              const nodeFlags = getCombinedNodeFlags(decl);
+              if (modifiers & ModifierFlags.Static) {
+                modifierSet |= 1 << TokenModifier.static;
+              }
+              if (modifiers & ModifierFlags.Async) {
+                modifierSet |= 1 << TokenModifier.async;
+              }
+              if (typeIdx !== TokenType.class && typeIdx !== TokenType.interface) {
+                if ((modifiers & ModifierFlags.Readonly) || (nodeFlags & NodeFlags.Const) || (symbol.getFlags() & SymbolFlags.EnumMember)) {
+                  modifierSet |= 1 << TokenModifier.readonly;
+                }
+              }
+              if ((typeIdx === TokenType.variable || typeIdx === TokenType.function) && isLocalDeclaration(decl, sourceFile)) {
+                modifierSet |= 1 << TokenModifier.local;
+              }
+              if (program.isSourceFileDefaultLibrary(decl.getSourceFile())) {
+                modifierSet |= 1 << TokenModifier.defaultLibrary;
+              }
+            } else if (symbol.declarations && symbol.declarations.some(d => program.isSourceFileDefaultLibrary(d.getSourceFile()))) {
+              modifierSet |= 1 << TokenModifier.defaultLibrary;
+            }
+
+            collector(node, typeIdx, modifierSet);
+
+          }
+        }
+      }
+      forEachChild(node, visit);
+
+      inJSXElement = prevInJSXElement;
+    }
+    visit(sourceFile);
+  }
+
+  function classifySymbol(symbol: Symbol, meaning: SemanticMeaning): TokenType | undefined {
+    const flags = symbol.getFlags();
+    if (flags & SymbolFlags.Class) {
+      return TokenType.class;
+    } else if (flags & SymbolFlags.Enum) {
+      return TokenType.enum;
+    } else if (flags & SymbolFlags.TypeAlias) {
+      return TokenType.type;
+    } else if (flags & SymbolFlags.Interface) {
+      if (meaning & SemanticMeaning.Type) {
+        return TokenType.interface;
+      }
```

Comment:
What case does this filter out? If this is the only place `getMeaningFromLocation` is used, is there a more direct way to work with the declaration for this case?